parser/types/field_type.go

```diff
@@ -90,6 +90,9 @@ func (ft *FieldType) IsVarLengthType() bool {
 
 // GetType returns the type of the FieldType.
 func (ft *FieldType) GetType() byte {
+	if ft.array {
+		return mysql.TypeJSON
```

Comment:
Should we add a `TypeArray`? Considering "whether it's a json or array" everywhere is a little tiring :thinking: (e.g. will `json_extract(cast(xxx as array), '$.a')` be allowed? Because many places don't consider the array situation, and I'm afraid it will lead to many unexpected branches)

Reply:
Not now, for this PR, we use JSON is fine. I will try to add `typeArray` in the latter PRs.
`cast(xxx as array)` can be only used in DDL, more specifically, in expression(functional) index, all other usages will report an error. You can see the `allowCastArray` in `planbuilder`